src/scripts/DeployToken.ts

```diff
@@ -0,0 +1,429 @@
+): Promise<{ salt: `0x${string}`; token: `0x${string}` }> {
+  const startingPoint = BigInt(
+    '0x' +
+      Array.from(crypto.getRandomValues(new Uint8Array(32)))
+        .map((b) => b.toString(16).padStart(2, '0'))
+        .join('')
+  );
+  let i = startingPoint;
+  while (true) {
+    const salt = `0x${i.toString(16).padStart(64, '0')}` as `0x${string}`;
+
+    const token = await predictToken_v3_1(
+      admin,
+      name,
+      symbol,
+      image,
+      metadata,
+      context,
+      supply,
+      salt,
+      initialSupplyChainId
+    );
+
+    const tokenNum = BigInt(token);
+    const pairedTokenNum = BigInt(pairedTokenAddress);
+
+    if (tokenNum < pairedTokenNum) {
+      return { salt, token };
+    }
+
+    i += BigInt(
+      Math.floor(crypto.getRandomValues(new Uint8Array(1))[0] % 1000) + 1
+    );
+  }
+}
+
+// Dummy transaction sender - replace with actual implementation (viem or ethers)
+async function sendTransactionAsync({
+  to,
+  data,
+  value,
+  gas,
+}: {
+  to: `0x${string}`;
+  data: `0x${string}`;
+  value: bigint;
+  gas: bigint;
+}): Promise<string> {
+  console.log('Sending transaction:', { to, data, value, gas });
+  // send transaction
+  return '0xTransactionHash';
+}
+
+/**
+ * Builds a token deployment transaction
+ */
+async function buildDeploymentTransaction({
+  deployerAddress,
+  formData,
+  chainId,
+  userInfo,
+  desiredPrice = 10, // Default desired price (pairs with WETH)
+}: {
+  deployerAddress: `0x${string}`;
+  formData: IDeployFormData;
+  chainId: number;
+  userInfo?: IUserInfo;
+  desiredPrice?: number;
+}): Promise<{
+  transaction: {
+    to: `0x${string}`;
+    data: `0x${string}`;
+  };
+  value: bigint;
+}> {
+  if (!deployerAddress) {
+    throw new Error('No account address found');
+  }
+
+  // Create metadata and social context
+  const clankerMetadata: IClankerMetadata = {
+    description: formData.description || 'No description provided',
+    socialMediaUrls: [
+      ...(formData.telegramLink
+        ? [{ platform: 'telegram', url: formData.telegramLink }]
+        : []),
+      ...(formData.websiteLink
+        ? [{ platform: 'website', url: formData.websiteLink }]
+        : []),
+      ...(formData.xLink ? [{ platform: 'x', url: formData.xLink }] : []),
+      ...(formData.farcasterLink
+        ? [{ platform: 'farcaster', url: formData.farcasterLink }]
+        : []),
+    ],
+    auditUrls: [],
+  };
+
+  const clankerSocialContext: IClankerSocialContext = {
+    interface: 'YOUR_INTERFACE_HERE',
+    platform: userInfo?.farcaster ? 'farcaster' : userInfo?.twitter ? 'X' : '',
+    messageId: '',
+    id: userInfo?.farcaster?.fid
+      ? userInfo.farcaster.fid.toString()
+      : userInfo?.twitter?.username || '',
+  };
+
+  // Determine paired token
+  const pairAddress =
+    formData.pairedToken && formData.pairedToken.length > 0
+      ? (formData.pairedToken as `0x${string}`)
+      : WETH_ADDRESS;
+
+  // Calculate initial tick for price
```

Comment:
can you add a note here that the contracts expect the tick to always be as if the created token has a lower address? (even when it's address is higher)